frontend/src/composables/use-sensitive-media.ts

```diff
@@ -0,0 +1,83 @@
+import { computed, watch } from "vue"
+
+import type { Media } from "~/types/media"
+
+import { useUiStore } from "~/stores/ui"
+
+type SensitiveVisibilityState =
+  | "insensitive"
+  | "sensitive-shown"
+  | "sensitive-hidden"
+
+/**
+ * A helper composable for working with sensitive media.
+ * Provides the current visibility of a sensitive media,
+ * along with toggles for hiding/revealing the media.
+ */
+export function useSensitiveMedia(
+  media: Pick<Media, "id" | "isSensitive"> | null
+) {
+  const uiStore = useUiStore()
+
+  // Clear out previously-revealed results when the toggle is switched.
+  watch(
+    () => uiStore.shouldBlurSensitive,
+    () => {
+      uiStore.revealedSensitiveResults = []
+    },
+    { deep: true }
```

Comment:
I tried to see if we could get rid of this `{ deep: true }`, because Pinia docs say it's expensive, and it is usually used to watch nested objects, and here it is a boolean. Well, apparently, a non-deep watcher is fired when the property is set to a new object, and that never happens to a boolean - I guess that's why `deep` is needed.

Then I added a `console.log` here to track when this watcher is called, and it's called about 27 times on the "cat" All content search page. So, it is called by **every** component that imports this composable. And we reset `revealedSensitiveResults` resultCount times every time we toggle "blur sensitive results".

I tried re-factoring this composable to be a `useSensitiveMedia` store, but I think a better solution would be to move this watcher to the `ui` store. We can simulate a watcher there by adding a `setShouldBlurSensitive` function, which would be toggling the property _and_ also resetting the `revealedSensitiveResults`. And never set the value directly, like `uiStore.shouldBlurSensitive="true"`.

I guess using watchers that watch global properties or store properties inside composables is a bad idea because those watchers would be triggered by every component that uses the composable.

Reply:
I noticed this last night, too, and think moving it to the UI store would be ideal. It's not a _super_ important piece of functionality, anyway, so definitely isn't worth the performance hit to have it here

Reply:
Resolved in e82bc7c43.